test/integration/spina/conferences/primer_theme/journal/issues_navigation_test.rb

```diff
@@ -29,6 +31,49 @@ class IssuesNavigationTest < ActionDispatch::IntegrationTest
             end
           end
 
+          test 'view issues when there are no issues' do
+            Spina::Admin::Journal::Issue.destroy_all
+            get frontend_issues_path
+            assert_response :success
+            assert_select 'div#journal-issues-list' do
+              assert_select 'h3', text: I18n.t('spina.conferences.primer_theme.journal.issues.index.no_issues')
+            end
+          end
+
+          test 'view issues when there are no past issues' do
+            get admin_logout_path
+
+            Spina::Admin::Journal::Issue.destroy_all
+            new_issue = Spina::Admin::Journal::Issue.create!(number: 1, volume: Spina::Admin::Journal::Volume.first, date: Time.zone.today + 1.day)
+            get frontend_issues_path
+            assert_response :success
+            assert_select 'div#journal-issues-list' do
+              assert_select 'h3', text: I18n.t('spina.conferences.primer_theme.journal.issues.index.no_issues')
+            end
+          end
+
+          test 'view issues as admin when there are no past issues' do
+            user = spina_users :joe
+            post admin_sessions_path, params: { email: user.email, password: 'password' }
```

Comment:
Can you refactor these lines into a private method (e.g. `log_in_as_admin`) and do the same for getting logout path? I think it will make it clearer what's going on.

Reply:
can do